---
title: "Use Case 1"
author: "Aidan MacNamara"
date: "29 March 2018"
output:
  html_document: default
  pdf_document: default
---

## Introduction ##

We have a starting gene list of possible targets that could modulate the upregulation of Type I Interferon signaling across a number of autoimmune diseases. We want to further interrogate these genes in a cell model. In this case, we have a choice between THP-1 and U937. We will use the **epiChoose** Shiny app to examine how these genes behave in the 2 cell lines and their relatedness to primary cell types from Blueprint.

```{r echo=FALSE, message=FALSE, warning=FALSE}
require(tidyverse)
require(knitr)
```


```{r echo=FALSE, message=FALSE, warning=FALSE}
my_genes = read_tsv("ifn_list_52.txt", col_names=FALSE)
print(my_genes$X1)
```

## Overview ## 

Before starting with a specific gene list, you might want to compare the candidate cell models and the primary data across the whole genome. This can be performed under the **Data Overview** tab. The THP-1 and U937 cell lines were profiled as part of **Project 2**. Hence, we can select Project 2 from the left panel. Since THP-1 and U937 are models of primary monocytes/macrophages, we will also select **Blueprint** from the left panel and manually select monocytes and macrophages from the text box underneath.

![](images/walkthrough_1.png)

After selecting the type of analysis you wish to perform (**PCA** or **MDS**), pressing **Run** produces multi-dimensiional scaling plots of the selected samples for each data type.

![](images/walkthrough_2.png)

The drop-down menu at the bottom can be used to select specific data types. In this case, by selecting H3K27ac, we can see a shift towards primary cell type patterns with the addition of PMA and LPS in both THP-1 and U937:

![](images/walkthrough_3.png)

## Specific Loci ##

To look at the differences at more specific loci, the **Data Exploration** tab can be selected. The selections on the left side take 3 categories:

1. **What method to use ...** Different methods of comparison. More details are available by hovering over the selection

2. **Cell Choice** This is split into 'Cell Target Choice' and 'Cell Model Choice' i.e. the choice of primary cells (or targets) and the possible cell models

3. **Gene Choice** The loci over which to make your comparisons can be entered manually ('Gene Choice'), by ontology ('Gene Ontology Choice'), or can be uploaded ('Choose your gene list file'). If more than 1 method is chosen, the gene choice is combined.

![](images/walkthrough_4.png)

Once the cell and gene selections have been made, **Update Parameters** can be selected (bottom left) and the selected plot will appear in the main panel. **Correlation** has been selected in the figure above. This gives the distance, in terms of Pearson correlation, between the cell models and the target cells for each data type i.e. the lower the cell model is on the y-axis, the closer it is to the target cells based on the similarity of signal for that data type at the loci selected. If more than 1 target cell is chosen (as in this case, where we have chosen a number of primary monocytes), the average value across the target cells will be used for the correlation.

Returning to the data, let's look at the suitability of THP-1 and U937 with a number of treatments (baseline/PMA/PMA+LPS) as models for primary CD14+/CD16- monocytes based on the uploaded gene list. For the **Correlation** plot above, we see that the U937 models seem to be closer to the primary monocytes based on the H3K27ac signal at the selected loci. H3K4me3 and H4K27me3 do not show any separation (and as we do not have ATAC-seq/CTCF/RNA-seq data currently available for the Blueprint samples, these do not show).

We can look at the correlation in the signal between 'U937_BR1_PMA+LPS' and the primary data in a little more detail by selecting **Scatterplot** on the left panel:

![](images/walkthrough_5.png)

**Scatterplot** plots data type scores on each axis (the data type can be selected on the right) for each selected sample. If, however, you choose **Same as x-axis** for the y-axis choice, it will plot correlations between samples instead. This is what the figure above shows - the correlation in H3K27ac signal for all sample combinations across the gene list. From this it is clear that 'U937 PMA+LPS' strongly correlates with the primary monocyte samples (relevant panels highlighted). And just to show that this is a cell-type specific result, below is the correlation of the primary monocytes with the lung model A549, which, as you would expect, is poorly correlated in terms of H3K27ac signal across this gene set:

![](images/walkthrough_6.png)

Looking at 'THP-1_BR1_PMA+LPS' using the same plots, the H3K27ac signal correlates well but with a couple of notable outliers, which can be selected by dragging a box around the points (shown in the bottom-left panel below):

![](images/walkthrough_7.png)

We can examine the signal of these 2 outlier genes - PTEN and CDK6, by selecting the **Browser** tab and entering the relevant data in the left panel. The resulting plot overlays the H3K27ac signal for each sample across the selected genes. As each data point from the correlation plot is the area under the curve (AUC) around the gene TSS, we can visually confirm the differences summarised in the correlation plot. There is a strong H3K27ac signal for 'THP-1_BR1_PMA+LPS' at the TSSs for CDK6 and PTEN, compared to the primary monocytes.

![](images/walkthrough_8.png)

## Current Gaps and Known Issues ##

1. We are still collecting data so comparisons are not currently available across all samples for all data types. Most notable is the missing expression data from Blueprint. This will be available shortly.

2. ATAC-seq and CTCF data types were not measured as part of Blueprint.

3. We are still working on the **Data Integration** and **Model Choice** tabs. We are investigating methods of integrating across data types to give a more accurate metric of relatedness between cells (as opposed to measuring each data type separately):

![](images/walkthrough_9.png)

4. The **Download** gives the metadata for all GSK samples covered in the project. Signal (bigwig) and BAM files will be available to download from here in the near future. For the moment, please send any data requests to aidan.x.macnamara@gsk.com.

![](images/walkthrough_10.png)

5. Any issues or feature requests can be added here: https://github.com/aidanmacnamara/epiView/issues

